src/api/world.ts

```diff
@@ -62,7 +62,6 @@ export const createFirestoreWorldStartInput: (
   const worldUpdateData: Partial<WithId<World>> = {
     ...omit(input, Object.keys(imageInputs)),
     ...imageInputData,
-    id,
```

Comment:
What happens if `generateFirestoreId()` gets invoked because of
```ts
  // NOTE: id is needed before world is created to upload the images
  const id = input?.id ?? generateFirestoreId({ emulated: true });
```
?
Can it ever be invoked?

Reply:
I don't thinks so, id should be always there.

Reply:
> Should

☝️ this is a big assumption. I will try to make another one, based on the code, before running and testing it.


Here is how the ID is being passed down in the FE:
```js
    const worldId = (
      await firebase.functions().httpsCallable("world-createWorld")(stubInput)
    )?.data;

    const fullInput = await createFirestoreWorldStartInput(
      withId(world, worldId),
      user
    );
```

And this is how this PR makes it always `undefined` :
```js
    .then(() => ({ ...worldData, id: worldDoc.id }));
```
@ https://github.com/sparkletown/sparkle/pull/2474/files#diff-c795148bf23657fc8fd09642f2b39a2789f97a4fe701fd8062eb9cd33f1c83caL89-R91

So `generateFirestoreId()` will likely always get executed and 
```js
    await firebase.functions().httpsCallable("world-updateWorld")(fullInput);
```
will try to update world other than the one created by `"world-createWorld"`, most likely with an empty string as an ID
```js
  let worldId = "";
```

I will try to test this.